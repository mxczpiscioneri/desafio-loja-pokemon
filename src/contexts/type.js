import React, { createContext, useContext } from 'react'
import PropTypes from 'prop-types'
import { useLocalStorage } from '../hooks/useLocalStorage'

const TypeContext = createContext({})

export const TypeProvider = ({ children }) => {
  const { storedValue, setValue } = useLocalStorage()

  return (
    <TypeContext.Provider value={{ type: storedValue.type, setType: setValue }}>
      {children}
    </TypeContext.Provider>
  )
}

export const useTypeContext = () => useContext(TypeContext)

TypeProvider.propTypes = {
  children: PropTypes.any.isRequired
}

export default TypeContext
