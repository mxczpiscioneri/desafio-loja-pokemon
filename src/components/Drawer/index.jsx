import React from 'react'
import PropTypes from 'prop-types'
import { SwipeableDrawer } from '@material-ui/core'
import Cart from '../Cart'

const Drawer = ({ open, toggleDrawer }) => {
  const iOS = /iPad|iPhone|iPod/.test(navigator.userAgent)
  const handlerOpenClose = (event) => {
    if (event) {
      event.preventDefault()
    }

    toggleDrawer()
  }

  return (
    <SwipeableDrawer
      anchor={'right'}
      open={open}
      onClose={handlerOpenClose}
      onOpen={handlerOpenClose}
      disableBackdropTransition={!iOS}
      disableDiscovery={iOS}
    >
      <Cart />
    </SwipeableDrawer>
  )
}

Drawer.propTypes = {
  open: PropTypes.bool.isRequired,
  toggleDrawer: PropTypes.func.isRequired
}

export default Drawer
