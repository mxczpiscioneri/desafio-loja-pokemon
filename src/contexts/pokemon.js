import React, { createContext, useContext, useState } from 'react'
import PropTypes from 'prop-types'
import { keys, useLocalStorage } from '../hooks/useLocalStorage'

const PokemonContext = createContext({})

export const PokemonProvider = ({ children }) => {
  const { storedValue, setValue } = useLocalStorage()
  const [drawer, setDrawer] = useState(false)

  const toggleDrawer = () => {
    setDrawer(status => !status)
  }

  const setType = (type) => {
    setValue(keys.type, type)
  }

  const addCart = (pokemon) => {
    const listPokemon = JSON.parse(storedValue.cart)
    listPokemon.push(pokemon)

    setValue(keys.cart, JSON.stringify(listPokemon))
    toggleDrawer()
  }

  const removeCart = (index) => {
    const listPokemon = JSON.parse(storedValue.cart)
    const newListPokemon = listPokemon.filter((_, i) => i !== index)

    setValue(keys.cart, JSON.stringify(newListPokemon))
  }

  const clearCart = () => {
    setValue(keys.cart, JSON.stringify([]))
  }

  return (
    <PokemonContext.Provider value={{
      type: storedValue.type,
      setType,
      cart: JSON.parse(storedValue.cart),
      addCart,
      removeCart,
      clearCart,
      drawer,
      toggleDrawer }}
    >
      {children}
    </PokemonContext.Provider>
  )
}

export const usePokemonContext = () => useContext(PokemonContext)

PokemonProvider.propTypes = {
  children: PropTypes.any.isRequired
}

export default PokemonContext
