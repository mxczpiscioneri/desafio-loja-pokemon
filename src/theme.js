import { createMuiTheme, responsiveFontSizes } from '@material-ui/core'

// colors
const primary = '#378B7D'
const primaryLight = '#639d25'
const primaryDark = '#427768'
const secondary = '#FA7D80'
const black = '#343a40'
const white = '#FFFFFF'
const darkBlack = 'rgb(36, 40, 44)'
const background = '#f5f5f5'
const backgroundMain = '#cfcfcf'
const warningLight = 'rgba(253, 200, 69, .3)'
const warningMain = 'rgba(253, 200, 69, .5)'
const warningDark = 'rgba(253, 200, 69, .7)'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primary,
      light: primaryLight,
      dark: primaryDark
    },
    secondary: { main: secondary },
    common: {
      white,
      black,
      darkBlack
    },
    warning: {
      light: warningLight,
      main: warningMain,
      dark: warningDark
    },
    background: {
      default: background,
      main: backgroundMain,
      gradient: `linear-gradient(45deg, ${primaryLight} 30%, ${primaryDark} 90%)`
    }
  },
  typography: {
    useNextVariants: true
  }
})

export default responsiveFontSizes(theme)
