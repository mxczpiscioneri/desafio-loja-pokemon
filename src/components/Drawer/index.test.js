import React from 'react'
import { render } from '@testing-library/react'
import { MuiThemeProvider } from '@material-ui/core'
import { ThemeProvider } from 'styled-components'
import Drawer from '.'
import theme from '../../theme'
import * as PokemonContext from '../../contexts/pokemon'

describe('Drawer components', () => {
  test('Render with success', () => {
    const mockContext = { cart: [] }
    jest
      .spyOn(PokemonContext, 'usePokemonContext')
      .mockImplementation(() => mockContext)

    const component = render(
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <Drawer open={true} toggleDrawer={jest.fn()} />
        </ThemeProvider>
      </MuiThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
