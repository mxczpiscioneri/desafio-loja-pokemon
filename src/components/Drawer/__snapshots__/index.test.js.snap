// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer components Render with success 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 0px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="PrivateSwipeArea-root-1 PrivateSwipeArea-anchorRight-3"
        style="width: 20px;"
      />
    </div>
    <div
      class="MuiDrawer-root MuiDrawer-modal"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorRight MuiPaper-elevation16"
        tabindex="-1"
      >
        <ul
          class="MuiList-root sc-bdfBwQ kBEneY MuiList-padding MuiList-subheader"
        >
          <li
            class="MuiListSubheader-root sc-gsTCUz bLgCqt MuiListSubheader-sticky MuiListSubheader-gutters"
          >
            Carrinho de compras
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Adicione um Pok√©mon no carrinho
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="PrivateSwipeArea-root-1 PrivateSwipeArea-anchorRight-3"
      style="width: 20px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
